extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultPKIManager.java

```diff
@@ -0,0 +1,615 @@
+        body.subjectAlternativeNames = options.subjectAlternativeNames != null
+                ? String.join(",", options.subjectAlternativeNames)
+                : null;
+        body.ipSubjectAlternativeNames = options.ipSubjectAlternativeNames != null
+                ? String.join(",", options.ipSubjectAlternativeNames)
+                : null;
+        body.uriSubjectAlternativeNames = options.uriSubjectAlternativeNames != null
+                ? String.join(",", options.uriSubjectAlternativeNames)
+                : null;
+        body.otherSubjectAlternativeNames = options.otherSubjectAlternativeNames;
+        body.keyType = options.keyType != null
+                ? options.keyType.name().toLowerCase()
+                : null;
+        body.keyBits = options.keyBits;
+        body.excludeCommonNameFromSubjectAlternativeNames = options.excludeCommonNameFromSubjectAlternativeNames;
+        body.subjectOrganization = options.subjectOrganization;
+        body.subjectOrganizationalUnit = options.subjectOrganizationalUnit;
+        body.subjectStreetAddress = options.subjectStreetAddress;
+        body.subjectPostalCode = options.subjectPostalCode;
+        body.subjectLocality = options.subjectLocality;
+        body.subjectProvince = options.subjectProvince;
+        body.subjectCountry = options.subjectCountry;
+        body.subjectSerialNumber = options.subjectSerialNumber;
+
+        VaultPKIGenerateIntermediateCSRResult internalResult = vaultInternalPKISecretEngine.generateIntermediateCSR(getToken(),
+                mount, type, body);
+
+        VaultPKIGenerateIntermediateCSRData internalResultData = internalResult.data;
+
+        GeneratedIntermediateCSRResult result = new GeneratedIntermediateCSRResult();
+        result.csr = internalResultData.csr;
+        result.privateKeyType = internalResultData.privateKeyType != null
+                ? CertificateKeyType.valueOf(internalResultData.privateKeyType.toUpperCase())
+                : null;
+        result.privateKey = internalResultData.privateKey;
+        return result;
+    }
+
+    @Override
+    public void setSignedIntermediateCA(String pemCert) {
+        VaultPKISetSignedIntermediateCABody body = new VaultPKISetSignedIntermediateCABody();
+        body.certificate = pemCert;
+
+        vaultInternalPKISecretEngine.setSignedIntermediateCA(getToken(), mount, body);
+    }
+
+    @Override
+    public void tidy(TidyOptions options) {
+        VaultPKITidyBody body = new VaultPKITidyBody();
+        body.tidyCertStore = options.tidyCertStore;
+        body.tidyRevokedCerts = options.tidyRevokedCerts;
+        body.safetyBuffer = options.safetyBuffer;
+
+        vaultInternalPKISecretEngine.tidy(getToken(), mount, body);
+    }
+
+    private void error(AbstractVaultDTO<?, ?> dto) {
+        if (dto.warnings instanceof List<?>) {
+            List<?> warnings = (List<?>) dto.warnings;
+            if (!warnings.isEmpty()) {
```

Comment:
No idea what else it could be 😔. The list was discovered via executing requests and seeing the results. The DTO wrapper lists `warnings` as `Object`. So, this is there because of that.

WRT, `else` branches, the `error` function will always throw an error. There is no escaping this function, it's only called once an error state is determined.

> Instead of "Unknown vault error", may be this should be something like no vault data because that is really what you are testing when you enter the error method.

This isn't true, although the test for `data == null` generally brings us here that isn't what we're testing. `data` is null when a warning has occurred.  All the warnings I've encountered are more properly describe as errors.

Finally, I'll implemented the `checkData` idea but will probably name it something along the lines of `checkWarning`.

Reply:
Changed mind... `checkWarnings` was not a good idea.. renamed to `checkDataValid` and now it includes the null check as requested.